pages/helpers.py

```diff
@@ -63,6 +63,20 @@ def get_extra(obj, attr_name, default=None):
         extra = obj.get('extra')
     return extra.get(attr_name, default) if extra else default
 
+
+def set_extra(obj, attr_name, attr_value):
+    prev_val = get_extra(obj, attr_name)
+
+    if not obj.extra:
+        # In case we have a None type or empty string, initialize
+        # extra as a dict because if we're using this method the intention
+        # is to set it to some value.
+        obj.extra = {}
+    obj.extra.update({attr_name: attr_value})
+    obj.extra = json.dumps(obj.extra)
```

Comment:
We don't want to serialize the extra field here (i.e. `json.dumps`) since the whole idea of `get_extra` is to unserialize it only once. Serialization should happen automatically on `obj.save()`.